src/metabase/api/database.clj

```diff
@@ -365,19 +373,43 @@
   (api/let-404 [db (Database id)]
     (api/write-check db)
     (db/delete! Database :id id)
-    (events/publish-event! :database-delete db))
+    (events/publish-event! :database-delete db)
+    (events/publish-event! :database-schedule-delete db))
   api/generic-204-no-content)
 
 
 ;;; ------------------------------------------------------------ POST /api/database/:id/sync ------------------------------------------------------------
 
+
 ;; TODO - Shouldn't we just check for superuser status instead of write checking?
+;; NOTE Atte: This becomes maybe obsolete
 (api/defendpoint POST "/:id/sync"
   "Update the metadata for this `Database`."
   [id]
   ;; just publish a message and let someone else deal with the logistics
   (events/publish-event! :database-trigger-sync (api/write-check Database id))
   {:status :ok})
 
+;; NOTE Atte Keinänen: If you think that these endpoints could have more descriptive names, please change them.
+;; Currently these match the titles of the admin UI buttons that call these endpoints
+
+;; Should somehow trigger sync-database/sync-database!
+(api/defendpoint POST "/:id/sync_schema"
+  "Trigger a manual update of the schema metadata for this `Database`."
+  [id]
+  {:status :ok})
+
+;; Should somehow trigger cached-values/cache-field-values-for-database!
+(api/defendpoint POST "/:id/rescan_values"
+  "Trigger a manual scan of the field values for this `Database`."
+  [id]
+  {:status :ok})
+
+;; "Discard saved field values" action in db UI
+(api/defendpoint POST "/:id/discard_values"
+  "Discards all saved field values for this `Database`."
+  [id]
+  {:status :ok})
```

Comment:
yup, that's what I was asking you to do =)

@attekei wrote some stubs to get roundtrips on the frontend working